### ISSUE TITLE
masks in match expression

### ISSUE THREAD

**Issue by chtenb** [mtoohey31/tree-sitter-koka#1]
I don't wanna pollute or keep the helix PR from being merged, so I'm creating an issue here instead.

![image](https://github.com/mtoohey31/tree-sitter-koka/assets/1030961/250d6070-9965-484d-ac2e-0b50ab679568)

Repro:

```koka
fun many-rec(p : parser<e,a>, input : sslice, acc : list<a>) : <div|e> (list<a>, sslice)
  match mask<div>
      parse(input, p)
    ParseOk(x, rest) -> many-rec(p, rest, Cons(x, acc))
    ParseError(_, _) -> (acc, input)

pub fun get() : e int { 42 }
```

I'm only starting to use masks now and I'm not very familiar with them, but this syntax was suggested to be valid by anfelor at https://github.com/koka-lang/koka/discussions/397#discussioncomment-8029792

**Comment by mtoohey31** [comment]
Thanks for opening this here. I'm a little busy with the start of the school semester, but I'll try to get this fixed soon!

**Comment by mtoohey31** [comment]
I agree that this syntax should be valid. It looks like there's an issue with indented expressions being recognized as blocks in the position used in your example. I've started trying to figure out a fix.

**Comment by mtoohey31** [comment]
Yet another case of the grammar "spec" being incorrect... it only allows "non-trailing lambda expressions" in match position, and while the expression in your example isn't one of those, the compiler still accepts it :sob:.

**Comment by mtoohey31** [comment]
Should be fixed as of 96d070c3700692858035f3524cc0ad944cef2594.

![2024-01-21T20:58:47,821903911-05:00](https://github.com/mtoohey31/tree-sitter-koka/assets/36740602/2e86b633-a78e-45a1-9718-72959f847c2b)

**Comment by chtenb** [comment]
> Yet another case of the grammar "spec" being incorrect... it only allows "non-trailing lambda expressions" in match position, and while the expression in your example isn't one of those, the compiler still accepts it 😭.

Perhaps you can open an issue in the koka repo for the discrepancies you found in the grammar spec. I think these ought to be fixed or flagged as undocumented compiler behavior.
